src/main.rs

```diff
@@ -0,0 +1,187 @@
+    ($fmt:expr, $($arg:tt)*) => (err_print!(concat!($fmt, "\n"), $($arg)*));
+}
+
+macro_rules! err_print {
+    ($($arg:tt)*) => (io::stderr().write_fmt(format_args!($($arg)*)).unwrap(););
+}
+
+macro_rules! create_replace_fn {
+    ($e:expr, $s:expr) => { match $e {
+        None => Box::new(|s: &str| String::from(s)),
+        Some(s) => {
+            match Regex::new(s.as_str()) {
+                Ok(re) => Box::new(move |s: &str| re.replace(s, "")),
+                Err(e) => {
+                    err_println!(concat!("ERROR: strip_", $s, ": {}"), e);
+                    return;
+                }
+            }
+        }
+    }}
+}
+
+// The create_replace_fn macro should use Box::new(String::from), but the compiler's type inference
+//  can't handle it. 
+#[allow(redundant_closure)]
+fn main() {
+    use ShowOption::*;
+
+    let args: Args = Docopt::new(USAGE)
+        .and_then(|d| d.decode())
+        .unwrap_or_else(|e| e.exit());
+
+    let fst_replace: Box<Fn(&str) -> String> = create_replace_fn!(args.flag_strip_fst, "fst");
+    let snd_replace: Box<Fn(&str) -> String> = create_replace_fn!(args.flag_strip_snd, "snd");
+
+    let (mut fst, mut snd) = match args.arg_name {
+        None => {
+            let fst = parse_benchmarks(File::open(&args.arg_file[0]).unwrap());
+            let snd = parse_benchmarks(File::open(&args.arg_file[1]).unwrap());
+
+            let fst = fst.map(|mut b| {
+                    b.name = fst_replace(b.name.as_str());
+                    b
+                })
+                .collect::<Vec<Benchmark>>();
+            let snd = snd.map(|mut b| {
+                    b.name = snd_replace(b.name.as_str());
+                    b
+                })
+                .collect::<Vec<Benchmark>>();
+
+            (fst, snd)
+        }
+        Some(ref names) => {
+            let parse_file = |s| parse_benchmarks(File::open(s).unwrap()).into_iter();
+
+            let benchmarks = args.arg_file.iter().flat_map(parse_file);
+
+            let mut fst = Vec::new();
+            let mut snd = Vec::new();
+
+            for mut b in benchmarks {
+                // explicitly moving the name out of b here so it can be assigned later
+                let name = b.name;
+                let mut split = name.splitn(2, "::");
+                // There should always be a string here
+                let implementation = split.next().unwrap();
+                // But there may not be a :: in the string, so the second part may not exist
+                if let Some(test) = split.next() {
+                    if implementation == &names[0] {
+                        b.name = fst_replace(test);
+                        fst.push(b);
+                    } else if implementation == &names[1] {
+                        b.name = snd_replace(test);
+                        snd.push(b);
+                    }
+                }
+            }
+
+            (fst, snd)
+        }
+    };
+
```

Comment:
It seems like it might be better to apply `fst_replace` and `snd_replace` here, where it only needs to be done once instead of repeating it in both branches above.